packages/smooth_app/lib/cache/files/files_cache_impl.dart

```diff
@@ -0,0 +1,123 @@
+// ignore_for_file: avoid_slow_async_io
+part of 'files_cache.dart';
+
+/// Implementation based on [getTemporaryDirectory]
+class _ShortLivingFileCacheManagerImpl extends _FileCacheManagerImpl {
+  _ShortLivingFileCacheManagerImpl._() : super._();
+
+  @override
+  Future<void> _init(String name) async {
+    return _initCache(name, await getTemporaryDirectory());
+  }
+}
+
+/// Implementation based on [getApplicationDocumentsDirectory]
+class _LongLivingFilesCacheManagerImpl extends _FileCacheManagerImpl {
+  _LongLivingFilesCacheManagerImpl._() : super._();
+
+  @override
+  Future<void> _init(String name) async {
+    return _initCache(name, await getApplicationDocumentsDirectory());
+  }
+}
+
+abstract class _FileCacheManagerImpl extends FileCache {
```

Comment:
Should not be abstract, but have two `final` fields `Directory rootFolder` and `String subFolderName` and one `Directory? _directory` (this one to be initialized `async`ly whenever needed).